src/useMercureSubscription.js

```diff
@@ -0,0 +1,34 @@
+import {
+  CRUD_GET_ONE_SUCCESS,
+  FETCH_END,
+  GET_ONE,
+  useDataProvider,
+} from 'ra-core';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+
+export default function useMercureSubscription(resource, idOrIds) {
+  const dataProvider = useDataProvider();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const ids = Array.isArray(idOrIds) ? idOrIds : [idOrIds];
+    dataProvider.subscribe(ids, (document) => {
```

Comment:
I think it could be nice to verify that the data provider used has the `subscribe` and `unsubscribe` methods, and log a message if it's not the case.
It will allow to use other data providers than the Hydra one.